homeassistant/components/soundtouch/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for Bose SoundTouch integration."""
+import logging
+
+from libsoundtouch import soundtouch_device
+from requests import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SoundtouchConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Bose SoundTouch."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize a new SoundTouch config flow."""
+        self.host = None
+        self.name = None
+
+    async def async_step_import(self, import_data):
+        """Handle a flow initiated by configuration file."""
+        self.host = import_data[CONF_HOST]
+        self.name = import_data[CONF_NAME]
+
+        try:
+            await self._async_get_device_id()
+        except RequestException:
+            return self.async_abort(reason="cannot_connect")
+
+        return await self._async_create_soundtouch_entry()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            self.host = user_input[CONF_HOST]
+            self.name = user_input[CONF_NAME]
+
+            try:
+                await self._async_get_device_id(raise_on_progress=False)
+            except RequestException:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self._async_create_soundtouch_entry()
+
+        return self.async_show_form(
+            step_id="user",
+            last_step=True,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): cv.string,
+                    vol.Optional(CONF_NAME, default=""): cv.string,
```

Comment:
Why do we need to ask for the name?

Reply:
I added the name because it was a property in the yaml configuration, but as this can be overridden for the media player entity via the GUI later, it should indeed be unnecessary. I will remove it.